### PR TITLE
Add debugging info during Zipkin export

### PR DESCRIPTION
**Pull request by andrewhood125**
Recently ran into an issue trying to integrate Zipkin and used this to help identify the issue. 

After 
Screen Shot 2022-01-03 at 4 14 37 PM

### CODE REVIEW

**Review by ericmustin — COMMENTED** [review]
Sorry for delay in reviewing 😬 . Anyway, I think broadly speaking there's a lot of room for improvement in how failed exporter response information is recorded and surfaced, and how easy that information is for a user to access. This PR improves what we currently have and so i'd like to see it merged. It does differ from the OTLP exporter, which uses the concept of an optional metric reporter for this https://github.com/open-telemetry/opentelemetry-ruby/blob/e58fe948a2552329f3cd64c5cf6256fdddbf427a/exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb#L53 which, i think has a lot of benefits but also is a bit cumbersome to setup. But, I don't think we should block adding debug logs to the zipkin exporter just because the otlp exporter does things differently.

However, would it be possible to only add this debug log for non 200 responses? ie before `redo` in `when` non-`SUCCESS` blocks? I worry that users in local dev will have on debug logging and find this library unnecessarily noisy when everything is working as expected (ie, `SUCCESS` responses so there's nothing to debug)

File: exporter/zipkin/lib/opentelemetry/exporter/zipkin/exporter.rb
Comment (andrewhood125) [reply exporter/zipkin/lib/opentelemetry/exporter/zipkin/exporter.rb]: I went ahead and dropped the response body.

**Comment by andrewhood125** [comment]
Thanks for the context, I haven't looked at the OTLP exporter but I will. I think logging for non success is the right way to go I've updated the PR.

File: exporter/zipkin/lib/opentelemetry/exporter/zipkin/exporter.rb
Comment (andrewhood125) [inline exporter/zipkin/lib/opentelemetry/exporter/zipkin/exporter.rb:172]: 👍 / 👎  on this block?

Comment (ericmustin) [reply exporter/zipkin/lib/opentelemetry/exporter/zipkin/exporter.rb:172]: hmmm, sure i think so? 👍

File: exporter/zipkin/lib/opentelemetry/exporter/zipkin/exporter.rb
Comment (ericmustin) [inline exporter/zipkin/lib/opentelemetry/exporter/zipkin/exporter.rb:151]: since we've moved the debug log into the non `SUCCESS` blocks and removed the `body` read from those calls, we should still keep this iiuc? other than that this lgtm

**Review by ericmustin — APPROVED** [review]
one small nit (i think we still need to keep the `response.body` read/discard in the main hot path since we no longer read it in the debug log) but other than that i think this lgtm, approving (assuming we making that change)

**Comment by andrewhood125** [comment]
Feedback addressed, thanks for the review!

**Comment by ericmustin** [comment]
@andrewhood125 (rubocop complaining about rubocop things)

**Comment by andrewhood125** [comment]
@ericmustin I believe this is ready to go now!

**Comment by ericmustin** [comment]
@andrewhood125 yup, lgtm and i've approved, cc @open-telemetry/ruby-maintainers

**Pull request closed without merging**
